widgets/search-box/search-box.js

```diff
@@ -65,36 +77,54 @@ function searchBox({
 
       // Add classes
       input.classList.add(cx(bem('input'), cssClasses.input));
+    },
+    addPoweredBy: function(input) {
+      var PoweredBy = require('../../components/PoweredBy/PoweredBy.js');
+      var poweredByContainer = document.createElement('div');
+      input.parentNode.insertBefore(poweredByContainer, input.nextSibling);
+      var poweredByCssClasses = {
+        root: cx(bem('powered-by'), cssClasses.poweredBy),
+        link: bem('powered-by-link')
+      };
+      ReactDOM.render(
+        <PoweredBy
+          cssClasses={poweredByCssClasses}
+        />,
+        poweredByContainer
+      );
+    },
+    init: function(initialState, helper) {
+      var isInputTargeted = container.tagName === 'INPUT';
+      var input = this.getInput();
 
+      // Add all the needed attributes and listeners to the input
+      this.addDefaultAttributesToInput(input, initialState.query);
       input.addEventListener('keyup', () => {
         helper.setQuery(input.value).search();
       });
 
+      if (isInputTargeted) {
+        // To replace the node, we need to create an intermediate node
+        var placeholderNode = document.createElement('div');
+        input.parentNode.insertBefore(placeholderNode, input);
+        input.parentNode.replaceChild(this.wrapInput(input), placeholderNode);
```

Comment:
I think it would be better to not call wrapInput here if the option is false. Versus calling it and checking the option inside the the function. Would make it clearer?